### PR TITLE
e2e_storage: stop using deprecated framework.ExpectEqual

### PR DESCRIPTION
**Pull request by Songjoy**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
e2e_storage: stop using deprecated framework.ExpectEqual
#### Which issue(s) this PR fixes:

Ref https://github.com/kubernetes/kubernetes/pull/115961

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by liangyuanpeng** [comment]
i can see that you're created much PR for it, thanks for your work and i'm really create a PR  before this PR: https://github.com/kubernetes/kubernetes/pull/119552 

This issue Great for my upfront contribution, I'd like to work on it, can you make it up to me? Thanks.

@Songjoy

**Comment by carlory** [comment]
Thanks for your contribution! @liangyuanpeng. @Songjoy has do a lot of work on this task. There're lots of similar issues in the codebase. This  link https://github.com/kubernetes/kubernetes/pull/119490/files#diff-f269a9e91aa6759831e67c7826ff093fa867167326f0b2c4a79967f54c126b86R649 is example about how to cleanup `framework.ExpectNotEqual`.

Could you help me to cleanup [other e2e tests](https://github.com/kubernetes/kubernetes/tree/master/test/e2e) in another prs. It's good first issue for new contributors. What do you think? If you want to continue to work on https://github.com/kubernetes/kubernetes/pull/119552, I think it's also ok and welcomed.

**Comment by Songjoy** [comment]
If you want to continue to work on #119552, I think it's also ok and welcomed. And I will close this PR.
@liangyuanpeng

File: test/e2e/storage/csi_mock/csi_volume_expansion.go [hunk 1]
```diff
@@ -537,7 +537,7 @@ func validateExpansionSuccess(ctx context.Context, pvc *v1.PersistentVolumeClaim
 	}
 
 	resizeStatus := pvc.Status.AllocatedResourceStatuses[v1.ResourceStorage]
-	framework.ExpectEqual(resizeStatus, "", "resize status should be empty")
+	gomega.Expect(resizeStatus).To(gomega.Equal(""), "resize status should be empty")
```
Comment (carlory) [inline test/e2e/storage/csi_mock/csi_volume_expansion.go:540]:
```suggestion
	gomega.Expect(resizeStatus).To(gomega.BeZero(), "resize status should be empty")
```

File: test/e2e/storage/csistoragecapacity.go [hunk 2]
```diff
@@ -152,29 +153,29 @@ var _ = utils.SIGDescribe("CSIStorageCapacity", func() {
...
-		framework.ExpectEqual(patchedCSC.Annotations["patched"], "true", "patched object should have the applied annotation")
+		gomega.Expect(patchedCSC.Annotations["patched"]).To(gomega.Equal("true"), "patched object should have the applied annotation")
 
 		ginkgo.By("updating")
 		csrToUpdate := patchedCSC.DeepCopy()
 		csrToUpdate.Annotations["updated"] = "true"
 		updatedCSC, err := cscClient.Update(ctx, csrToUpdate, metav1.UpdateOptions{})
 		framework.ExpectNoError(err)
-		framework.ExpectEqual(updatedCSC.Annotations["updated"], "true", "updated object should have the applied annotation")
+		gomega.Expect(updatedCSC.Annotations["updated"]).To(gomega.Equal("true"), "updated object should have the applied annotation")
```
Comment (carlory) [inline test/e2e/storage/csistoragecapacity.go:178]: and here.

File: test/e2e/storage/pd.go [hunk 3]
```diff
@@ -403,17 +404,17 @@ var _ = utils.SIGDescribe("Pod Disks [Feature:StorageProvider]", func() {
 					framework.ExpectNoError(err, fmt.Sprintf("Unable to create gcloud client err=%v", err))
 					output, err := gceCloud.ListInstanceNames(framework.TestContext.CloudConfig.ProjectID, framework.TestContext.CloudConfig.Zone)
 					framework.ExpectNoError(err, fmt.Sprintf("Unable to get list of node instances err=%v output=%s", err, output))
-					framework.ExpectEqual(true, strings.Contains(string(output), string(host0Name)))
+					gomega.Expect(true).To(gomega.Equal(strings.Contains(string(output), string(host0Name))))
```
Comment (carlory) [inline test/e2e/storage/pd.go:407]: please use ContainsSubstring instead.

File: test/e2e/storage/pd.go [hunk 4]
```diff
@@ -403,17 +404,17 @@ var _ = utils.SIGDescribe("Pod Disks [Feature:StorageProvider]", func() {
...
 					err = gceCloud.DeleteInstance(framework.TestContext.CloudConfig.ProjectID, framework.TestContext.CloudConfig.Zone, string(host0Name))
 					framework.ExpectNoError(err, fmt.Sprintf("Failed to delete host0Pod: err=%v", err))
 					ginkgo.By("expecting host0 node to be re-created")
 					numNodes := countReadyNodes(ctx, cs, host0Name)
-					framework.ExpectEqual(numNodes, origNodeCnt, fmt.Sprintf("Requires current node count (%d) to return to original node count (%d)", numNodes, origNodeCnt))
+					gomega.Expect(numNodes).To(gomega.Equal(origNodeCnt), fmt.Sprintf("Requires current node count (%d) to return to original node count (%d)", numNodes, origNodeCnt))
 					output, err = gceCloud.ListInstanceNames(framework.TestContext.CloudConfig.ProjectID, framework.TestContext.CloudConfig.Zone)
 					framework.ExpectNoError(err, fmt.Sprintf("Unable to get list of node instances err=%v output=%s", err, output))
-					framework.ExpectEqual(true, strings.Contains(string(output), string(host0Name)))
+					gomega.Expect(true).To(gomega.Equal(strings.Contains(string(output), string(host0Name))))
```
Comment (carlory) [inline test/e2e/storage/pd.go:417]: and here.

File: test/e2e/storage/persistent_volumes-gce.go [hunk 5]
```diff
@@ -20,6 +20,7 @@ import (
 	"context"
 
 	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes-gce.go:23]: please leave an empty line between L23 and L24.

File: test/e2e/storage/persistent_volumes-gce.go [hunk 6]
```diff
@@ -130,7 +131,7 @@ var _ = utils.SIGDescribe("PersistentVolumes GCEPD [Feature:StorageProvider]", f
 
 		ginkgo.By("Deleting the Claim")
 		framework.ExpectNoError(e2epv.DeletePersistentVolumeClaim(ctx, c, pvc.Name, ns), "Unable to delete PVC ", pvc.Name)
-		framework.ExpectEqual(verifyGCEDiskAttached(diskName, node), true)
+		gomega.Expect(verifyGCEDiskAttached(diskName, node)).To(gomega.BeTrue())
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes-gce.go:134]: Can you change this to not use BeTrue? See https://github.com/kubernetes/kubernetes/issues/105678.

File: test/e2e/storage/persistent_volumes-gce.go [hunk 7]
```diff
@@ -145,7 +146,7 @@ var _ = utils.SIGDescribe("PersistentVolumes GCEPD [Feature:StorageProvider]", f
 
 		ginkgo.By("Deleting the Persistent Volume")
 		framework.ExpectNoError(e2epv.DeletePersistentVolume(ctx, c, pv.Name), "Failed to delete PV ", pv.Name)
-		framework.ExpectEqual(verifyGCEDiskAttached(diskName, node), true)
+		gomega.Expect(verifyGCEDiskAttached(diskName, node)).To(gomega.BeTrue())
```
Comment (carlory) [inline test/e2e/storage/persistent_volumes-gce.go:149]: and here.

File: test/e2e/storage/pv_protection.go [hunk 8]
```diff
@@ -89,7 +90,7 @@ var _ = utils.SIGDescribe("PV Protection", func() {
 		ginkgo.By("Checking that PV Protection finalizer is set")
 		pv, err = client.CoreV1().PersistentVolumes().Get(ctx, pv.Name, metav1.GetOptions{})
 		framework.ExpectNoError(err, "While getting PV status")
-		framework.ExpectEqual(slice.ContainsString(pv.ObjectMeta.Finalizers, volumeutil.PVProtectionFinalizer, nil), true, "PV Protection finalizer(%v) is not set in %v", volumeutil.PVProtectionFinalizer, pv.ObjectMeta.Finalizers)
+		gomega.Expect(slice.ContainsString(pv.ObjectMeta.Finalizers, volumeutil.PVProtectionFinalizer, nil)).To(gomega.BeTrue(), "PV Protection finalizer(%v) is not set in %v", volumeutil.PVProtectionFinalizer, pv.ObjectMeta.Finalizers)
```
Comment (carlory) [inline test/e2e/storage/pv_protection.go:93]: please use ContainsSubstring instead.

File: test/e2e/storage/pvc_protection.go [hunk 9]
```diff
@@ -105,7 +106,7 @@ var _ = utils.SIGDescribe("PVC Protection", func() {
 		ginkgo.By("Checking that PVC Protection finalizer is set")
 		pvc, err = client.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(ctx, pvc.Name, metav1.GetOptions{})
 		framework.ExpectNoError(err, "While getting PVC status")
-		framework.ExpectEqual(slice.ContainsString(pvc.ObjectMeta.Finalizers, volumeutil.PVCProtectionFinalizer, nil), true, "PVC Protection finalizer(%v) is not set in %v", volumeutil.PVCProtectionFinalizer, pvc.ObjectMeta.Finalizers)
+		gomega.Expect(slice.ContainsString(pvc.ObjectMeta.Finalizers, volumeutil.PVCProtectionFinalizer, nil)).To(gomega.BeTrue(), "PVC Protection finalizer(%v) is not set in %v", volumeutil.PVCProtectionFinalizer, pvc.ObjectMeta.Finalizers)
```
Comment (carlory) [inline test/e2e/storage/pvc_protection.go:109]: BeElementOf

File: test/e2e/storage/pvc_storageclass.go [hunk 10]
```diff
@@ -22,6 +22,7 @@ import (
 	"time"
 
 	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
 	v1 "k8s.io/api/core/v1"
```
Comment (carlory) [inline test/e2e/storage/pvc_storageclass.go:26]: please leave an empty

File: test/e2e/storage/testsuites/provisioning.go [hunk 11]
```diff
@@ -938,7 +938,7 @@ func PVWriteReadSingleNodeCheck(ctx context.Context, client clientset.Interface,
 //
 // This is a common test that can be called from a StorageClassTest.PvCheck.
 func PVMultiNodeCheck(ctx context.Context, client clientset.Interface, timeouts *framework.TimeoutContext, claim *v1.PersistentVolumeClaim, node e2epod.NodeSelection) {
-	framework.ExpectEqual(node.Name, "", "this test only works when not locked onto a single node")
+	gomega.Expect(node.Name).To(gomega.Equal(""), "this test only works when not locked onto a single node")
```
Comment (carlory) [inline test/e2e/storage/testsuites/provisioning.go:941]:
```suggestion
	gomega.Expect(node.Name).To(gomega.BeZero(), "this test only works when not locked onto a single node")
```

File: test/e2e/storage/testsuites/snapshottable.go [hunk 12]
```diff
@@ -440,9 +440,9 @@ func checkSnapshot(ctx context.Context, dc dynamic.Interface, sr *storageframewo
 	ginkgo.By("checking the SnapshotContent")
 	// PreprovisionedCreatedSnapshot do not need to set volume snapshot class name
 	if pattern.SnapshotType != storageframework.PreprovisionedCreatedSnapshot {
-		framework.ExpectEqual(snapshotContentSpec["volumeSnapshotClassName"], vsc.GetName())
+		gomega.Expect(snapshotContentSpec["volumeSnapshotClassName"]).To(gomega.Equal(vsc.GetName()))
 	}
-	framework.ExpectEqual(volumeSnapshotRef["name"], vs.GetName())
-	framework.ExpectEqual(volumeSnapshotRef["namespace"], vs.GetNamespace())
+	gomega.Expect(volumeSnapshotRef["name"]).To(gomega.Equal(vs.GetName()))
```
Comment (carlory) [inline test/e2e/storage/testsuites/snapshottable.go:445]: and here.

File: test/e2e/storage/utils/utils.go [hunk 13]
```diff
@@ -141,14 +140,14 @@ func TestVolumeUnmountsFromDeletedPodWithForceOption(ctx context.Context, c clie
 	result, err := e2essh.SSH(ctx, fmt.Sprintf("mount | grep %s | grep -v volume-subpaths", clientPod.UID), nodeIP, framework.TestContext.Provider)
 	e2essh.LogResult(result)
 	framework.ExpectNoError(err, "Encountered SSH error.")
-	framework.ExpectEqual(result.Code, 0, fmt.Sprintf("Expected grep exit code of 0, got %d", result.Code))
+	gomega.Expect(result.Code).To(gomega.Equal(0), fmt.Sprintf("Expected grep exit code of 0, got %d", result.Code))
```
Comment (carlory) [inline test/e2e/storage/utils/utils.go:143]:
```suggestion
	gomega.Expect(result.Code).To(gomega.BeZero(), fmt.Sprintf("Expected grep exit code of 0, got %d", result.Code))
```

File: test/e2e/storage/vsphere/vsphere_volume_diskformat.go [hunk 14]
```diff
@@ -147,7 +147,7 @@ func invokeTest(ctx context.Context, f *framework.Framework, client clientset.In
 	framework.ExpectNoError(err)
 
 	ginkgo.By("Verify Disk Format")
-	framework.ExpectEqual(verifyDiskFormat(ctx, client, nodeName, pv.Spec.VsphereVolume.VolumePath, diskFormat), true, "DiskFormat Verification Failed")
+	gomega.Expect(verifyDiskFormat(ctx, client, nodeName, pv.Spec.VsphereVolume.VolumePath, diskFormat)).To(gomega.BeTrue(), "DiskFormat Verification Failed")
```
Comment (carlory) [inline test/e2e/storage/vsphere/vsphere_volume_diskformat.go:150]: and here.

File: test/e2e/storage/vsphere/vsphere_volume_disksize.go [hunk 15]
```diff
@@ -21,6 +21,7 @@ import (
 	"time"
 
 	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
```
Comment (carlory) [inline test/e2e/storage/vsphere/vsphere_volume_disksize.go:24]: and here.

File: test/e2e/storage/vsphere/vsphere_volume_node_delete.go [hunk 16]
```diff
@@ -81,7 +81,7 @@ var _ = utils.SIGDescribe("Node Unregister [Feature:vsphere] [Slow] [Disruptive]
 
 		// Ready nodes should be 1 less
 		ginkgo.By("Verifying the ready node counts")
-		framework.ExpectEqual(verifyReadyNodeCount(ctx, f.ClientSet, totalNodesCount-1), true, "Unable to verify expected ready node count")
+		gomega.Expect(verifyReadyNodeCount(ctx, f.ClientSet, totalNodesCount-1)).To(gomega.BeTrue(), "Unable to verify expected ready node count")
```
Comment (carlory) [inline test/e2e/storage/vsphere/vsphere_volume_node_delete.go:84]: and here.

File: test/e2e/storage/vsphere/vsphere_volume_node_delete.go [hunk 17]
```diff
@@ -98,7 +98,7 @@ var _ = utils.SIGDescribe("Node Unregister [Feature:vsphere] [Slow] [Disruptive]
 
 		// Ready nodes should be equal to earlier count
 		ginkgo.By("Verifying the ready node counts")
-		framework.ExpectEqual(verifyReadyNodeCount(ctx, f.ClientSet, totalNodesCount), true, "Unable to verify expected ready node count")
+		gomega.Expect(verifyReadyNodeCount(ctx, f.ClientSet, totalNodesCount)).To(gomega.BeTrue(), "Unable to verify expected ready node count")
```
Comment (carlory) [inline test/e2e/storage/vsphere/vsphere_volume_node_delete.go:101]: and here.

**Review by pohly — COMMENTED** [review]
@Songjoy thanks for the PR and thanks @carlory for the review and suggestions.

/approve

**Pull request merged**